pkg/adapter/modeldata/modeldata.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package modeldata contains the following model data in gnmi proto struct:
+//	openconfig-interfaces 2.0.0,
+//	openconfig-openflow 0.1.0,
+//	openconfig-platform 0.5.0,
+//	openconfig-system 0.2.0.
+package modeldata
```

Comment:
For the Unit tests you could test with `TestDevice-1.0.0` which have a very wide range of YANG constructs. You could depend on this like
```
	td1 "github.com/onosproject/config-models/modelplugin/testdevice-1.0.0/testdevice_1_0_0"
	td2 "github.com/onosproject/config-models/modelplugin/testdevice-2.0.0/testdevice_2_0_0"
```

Reply:
Regarding the test models, I would prefer to hold off on this for the time being.
I agree that the test yang and generated model  is a useful resource; it certainly covers more than the junos fragment that we are currently using.
However, I can see that it would expose some of the known limitations in the current adapter implementation, relating to data type transformations; these are marked by TODOs in the code.
The TODOs will obviously need to be addressed in due course, but:
	•	For the time being I don’t think they compromise the value of the initial test/demo
	•	When they are implemented, I’d prefer to have access to a real schema/device combination to inform and validate the implementation